spec/controllers/dashboard/tags_controller_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'rails_helper'
+
+RSpec.describe Dashboard::TagsController, type: :controller do
+  let(:tag) { create(:tag) }
+  let(:valid_attributes) { attributes_for(:tag) }
+  let(:invalid_attributes) { { name: '' } }
+
+  shared_examples 'user do unauthorize action' do |action|
+    it 'redirects to root path' do
+      case action
+      when :index
+        get action
+      when :create
+        post action, params: { tag: valid_attributes }, format: :js
+      when :update
+        put action, params: { id: tag.id, tag: valid_attributes }, format: :js
+      when :destroy
+        delete action, params: { id: tag.id }, format: :js
+      end
+
+      should redirect_to root_path
+      should set_flash[:alert]
+    end
+  end
+
+  describe 'GET #index' do
```

Comment:
Missing GET #new #edit #show, should check reponse of all those actions too

Reply:
Actually, the `Dashboard::TagsController` did not have these action you reminded, but I just realize that I have a `TagsController` which have two action #index and #show i have not tested, I have write it in the last commit, thank you for reminding me.